detector.go

```diff
@@ -373,47 +571,66 @@ func (bg BuildpackGroup) append(group ...BuildpackGroup) BuildpackGroup {
 
 type BuildpackOrder []BuildpackGroup
 
-func (bo BuildpackOrder) Detect(c *DetectConfig) (BuildpackGroup, BuildPlan, error) {
+type DetectResult struct {
+	BuildGroup           BuildpackGroup
+	BuildPrivilegedGroup BuildpackGroup
+	BuildPlan            BuildPlan
+	RunGroup             BuildpackGroup
+	RunPlan              BuildPlan
+}
+
+func (bo BuildpackOrder) Detect(c *DetectConfig) (*DetectResult, error) {
 	if c.runs == nil {
 		c.runs = &sync.Map{}
 	}
-	bps, entries, err := bo.detect(nil, nil, false, &sync.WaitGroup{}, c)
+
+	dr, err := bo.detect(nil, nil, false, &sync.WaitGroup{}, c)
 	if err == errBuildpack {
 		err = NewLifecycleError(err, ErrTypeBuildpack)
 	} else if err == errFailedDetection {
 		err = NewLifecycleError(err, ErrTypeFailedDetection)
 	}
-	for i := range entries {
-		for j := range entries[i].Requires {
-			entries[i].Requires[j].convertVersionToMetadata()
+
+	if dr != nil {
```

Comment:
When might `dr` be nil?

Reply:
`dr` is now null in case of `errFailedDetection` or `errBuildpack` (and probably other situations in the recursion). `detect` is returning a pointer of `DetectResult`

Reply:
Hmm... then couldn't you do something like 

```
	if err == errBuildpack {
		return nil, NewLifecycleError(err, ErrTypeBuildpack)
```

etc.?

Unless there are cases when `dr` can be nil without an error having occurred?